sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StatefulProcessorHandleImpl.scala

```diff
@@ -166,4 +169,45 @@ class StatefulProcessorHandleImpl(
       procTimers.remove(expiryTimestampMs)
     }
   }
+
+  /**
+   * Function to register a event time timer for given implicit key
+   *
+   * @param expiryTimestampMs - timer expiry timestamp in milliseconds
+   */
+  override def registerEventTimeTimer(expiryTimestampMs: Long): Unit = {
+    verify(timeoutMode == EventTime, s"Cannot register event time " +
+      "timers with incorrect TimeoutMode")
+    verify(currState == INITIALIZED || currState == DATA_PROCESSED,
+      s"Cannot register event time timer with " +
+        s"expiryTimestampMs=$expiryTimestampMs in current state=$currState")
+
+    if (eventTimers.exists(expiryTimestampMs)) {
+      logWarning(s"Timer already exists for expiryTimestampMs=$expiryTimestampMs")
```

Comment:
Lets separate these out into proc time and event time timers ? Could we also log the `queryId/runId` here ?

Reply:
Sorry, what do you mean by separate this out? Aren't they already in separate methods? Do you mean in what/how we log?

Reply:
I mean lets update the message to reflect this is a event-time based timer. Also update the message on the proc-time based timer logs too